WebDriverAgentMac/WebDriverAgentLib/Commands/FBScreenshotCommands.m

```diff
@@ -39,4 +43,31 @@ + (NSArray *)routes
   return FBResponseWithObject(screenshot);
 }
 
++ (id<FBResponsePayload>)handleGetScreenshots:(FBRouteRequest *)request
+{
+  NSNumber *desiredId = request.arguments[@"displayId"];
+  NSMutableDictionary <NSNumber *, NSDictionary *> *result = [NSMutableDictionary new];
+  for (XCUIScreen *screen in XCUIScreen.screens) {
+    NSNumber *displayId = [screen valueForKey:@"_displayID"];
+    if (nil != desiredId && (nil == displayId || ![desiredId isEqualToNumber:displayId])) {
+      continue;
+    }
+
+    NSNumber *isMainDisplay = [screen valueForKey:@"_isMainScreen"];
+    NSData *screenshotData = screen.screenshot.PNGRepresentation;
+    NSDictionary *screenInfo = @{
+      @"id": displayId ?: NSNull.null,
+      @"isMain": isMainDisplay ?: NSNull.null,
+      @"payload": screenshotData == nil ? NSNull.null : [screenshotData base64EncodedStringWithOptions:NSDataBase64Encoding64CharacterLineLength]
+    };
+    [result setObject:screenInfo forKey:displayId];
```

Comment:
should be `forKey:displayId.stringValue`, or what about handling `desiredId` as NSString and compare the value as string instead of Number?

It returned below response:
```
Object { error: "unknown error", message: "Invalid (non-string) key in JSON dictionary", traceback: "(\n\t0 CoreFoundation 0x00007fff34eae4d7 __exceptionPreprocess + 250\n\t1 libobjc.A.dylib 0x00007fff6da275bf objc_exception_throw + 48\n\t2 Foundation 0x00007fff37541155 ___writeJSONObject_block_invoke_2 + 111\n\t3 CoreFoundation 0x00007fff34df6587 __NSDICTIONARY_IS_CALLING_OUT_TO_A_BLOCK__ + 7\n\t4 CoreFoundation 0x00007fff34e3559f -[__NSDictionaryM enumerateKeysAndObjectsWithOptions:using…dReturnError:] + 615\n\t70 XCTest 0x0000000100260a7e -[XCTestDriver _runTests] + 466\n\t71 XCTest 0x0000000100304b82 _XCTestMain + 108\n\t72 WebDriverAgentRunner-Runner 0x0000000100003d41 _XCTRunnerRunTests + 25\n\t73 WebDriverAgentRunner-Runner 0x0000000100003d28 _XCTRunnerRunTests + 0\n\t74 libdyld.dylib 0x00007fff6ebcfcc9 start + 1\n\t75 ??? 0x0000000000000005 0x0 + 5\n)" }

```
Xcode 11, macOS 10.15.7